packages/teams-js/src/private/externalAppDialog.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { sendMessageToParent } from '../internal/communication';
+import { registerHandler, removeHandler } from '../internal/handlers';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { BaseDialogInfo } from '../public/interfaces';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ *  capability that allows an app (restricted to bizchat for now) to show a modal dialog.
+ *  Unlike traditional URL-based dialogs that limit opening URLs with the same domain,
+ *  this capability allows ME that has a different url domain to be opened within the dialog.
+ *
+ * It also allows chaining
+ */
+export namespace externalAppDialog {
+  export interface ISdkResponse {
+    err?: string;
+    result?: string | object;
+  }
+
+  export enum DialogContentType {
+    Card = 'card',
+    Url = 'url',
+    Text = 'text',
+  }
+
+  /**
+   * Base of a discriminated union between a URL based, an Adaptive Card and Text based Dialog info
+   */
+  interface DialogBase<T extends DialogContentType> extends BaseDialogInfo {
+    type: T;
+  }
+
+  /**
+   * URL based Dialog info
+   */
+  interface DialogWithUrl extends DialogBase<DialogContentType.Url> {
+    url: string;
+    appId: string;
+  }
+  interface DialogWithAdaptiveCard extends DialogBase<DialogContentType.Card> {
+    card: string;
+  }
+  interface DialogWithText extends DialogBase<DialogContentType.Text> {
+    text: string;
+  }
+  interface CloseDialog {
+    type: 'close';
+  }
+  export type ExternalAppDialogInfo = DialogWithUrl | DialogWithAdaptiveCard | DialogWithText;
+  export type ChainDialogInfo = CloseDialog | ExternalAppDialogInfo;
+
+  export type DialogSubmitHandler = (result: ISdkResponse) => ChainDialogInfo;
+
+  export function open(externalAppDialogInfo: ExternalAppDialogInfo, submitHandler: DialogSubmitHandler): void {
```

Comment:
I think the `ExternalAppDialogInfo` that is passed to `open` needs to be different from the one that makes up the `ChainDialogInfo`. Specifically, I think the `open` needs to include the height and width of the dialog, but the `ChainDialogInfo` should not.